### PR TITLE
Added google firebase authentication

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Created a Login page
Added google firebase authentication
image

### CODE REVIEW

**Comment by lukechurch** [comment]
I think you'll also need to adjust rules.json at the same time as this? You can do that by changing it in this PR

**Comment by IsaackMwenda** [comment]
I have Added firestore.rules
Fixed page flickering on data update
Fixed identation and whitespaces
Added a firebase.js file containing firebase API statements
Added a favicon

**Comment by IsaackMwenda** [comment]
Added columns for wrong scheme and not codable messages i.e (Number of messages and %)
image

**Comment by lukechurch** [comment]
> Added columns for wrong scheme and not codable messages i.e (Number of messages and %)
> image

Thanks very much for this @IsaackMwenda, impressive! - however this is conceptually separate from authentication so in future on a separate branch and PR please. (cc @ritazagoni )

File: public/coda/firestore.rules [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        function getDatasetResource(){
+            return get(/databases/$(database)/documents/$(documents))
+        }
+        allow read: if userIsAllowed(resource)
+      match /metrics/{metrics=**} {
+      	allow read, write: if userIsAllowed(getDatasetResource())
+      }
+    }
+  }
+
```
Comment (lukechurch) [inline public/coda/firestore.rules]: Sorry @IsaackMwenda I don't think this works, I'm going to remove it from the PR

**Review by lukechurch — APPROVED** [review]

**Pull request merged**